drivers/ma_yya_report/app/models/ma_yya_report/universe_calculator.rb

```diff
@@ -0,0 +1,206 @@
+      enrollments_by_client_id = clients_with_enrollments(batch)
+      {}.tap do |clients|
+        batch.each do |client|
+          client_id = client.id
+          enrollment = enrollments_by_client_id[client_id].last
+          enrollment_cls = enrollment.enrollment.current_living_situations.detect { |cls| cls.InformationDate == enrollment.first_date_in_program }
+          education_status = enrollment.enrollment.youth_education_statuses.max_by(&:InformationDate)
+          health_and_dv = enrollment.enrollment.health_and_dvs.max_by(&:InformationDate)
+
+          clients[client_id] = Client.new(
+            client_id: client_id,
+            service_history_enrollment_id: enrollment.id,
+            entry_date: enrollment.first_date_in_program,
+            referral_source: enrollment.enrollment.ReferralSource,
+            currently_homeless: currently_homeless?(enrollment_cls),
+            at_risk_of_homelessness: at_risk_of_homelessness?(enrollment_cls),
+            initial_contact: initial_contact(enrollments_by_client_id[client_id]),
+            direct_assistance: direct_assistance?(enrollment.enrollment),
+            education_status_date: education_status&.InformationDate,
+            current_school_attendance: education_status&.CurrentSchoolAttend,
+            current_educational_status: education_status&.CurrentEdStatus,
+            age: enrollment.client.age_on([@filter.start_date, enrollment.first_date_in_program].max),
+            gender: gender(enrollment.client),
+            race: race(enrollment.client),
+            ethnicity: enrollment.client.Ethnicity,
+            mental_health_disorder: disability?(enrollment.enrollment, 9),
+            substance_use_disorder: disability?(enrollment.enrollment, 10, [1, 2, 3]),
+            physical_disability: disability?(enrollment.enrollment, 5) || disability?(enrollment.enrollment, 7),
+            developmental_disability: disability?(enrollment.enrollment, 6),
+            pregnant: health_and_dv&.PregnancyStatus,
+            due_date: health_and_dv&.DueDate,
+            head_of_household: enrollment.head_of_household?,
+            household_ages: household_ages(enrollment),
+            sexual_orientation: enrollment.enrollment.SexualOrientation,
+            most_recent_education_status: education_status&.MostRecentEdStatus,
+            health_insurance: enrollment.enrollment.income_benefits_at_entry&.InsuranceFromAnySource == 1,
+            rehoused_on: rehoused_on(enrollment.enrollment),
+            subsequent_current_living_situations: subsequent_current_living_situations(enrollment.enrollment),
+            zip_code: zip_code(client),
+            flex_funds: flex_funds(client),
+          )
+        end
+      end
+    end
+
+    private def client_scope
+      enrollment_scope = GrdaWarehouse::ServiceHistoryEnrollment.open_between(start_date: @filter.start_date, end_date: @filter.end_date)
+      enrollment_scope = filter_for_projects(enrollment_scope)
```

Comment:
I don't particularly like that the internal variable has the same name as a private method.  maybe just `enrollments` in here?

Reply:
or dry this up with an enrollment_scope and an enrollment_with_preloads scope.  The logic looks the same, though this is missing `.entry`